src/realm/query_engine.cpp

```diff
@@ -51,6 +51,9 @@ size_t ParentNode::find_first(size_t start, size_t end)
 
 void ParentNode::aggregate_local_prepare(Action TAction, DataType col_id, bool nullable)
 {
+    // Verify that the cached column accessor is still valid
+    verify_column(); // throws
```

Comment:
Good idea putting the check in the init function. I don't know why I did not get that idea in the first place. It seams quite obvious.
I am well aware that the test coverage is not that great. Especially in the light of the problems that you have discovered. On the other hand I think a test like the one Thomas made would be a bit overkill in this case, so I hope that the combination of some test cases and review will do the job in this case.